src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/RouteAttributeHelper.cs

```diff
@@ -37,6 +37,18 @@ public static void SetRouteAttributeTag(this Activity activity, RouteEndpoint en
 
     private static string GetRoutePattern(RoutePattern routePattern, RouteValueDictionary routeValues)
     {
+        if (routePattern.PathSegments.Count == 0)
+        {
+            // RazorPage default route
+            if (routePattern.Defaults.ContainsKey("page"))
+            {
+                return routePattern.Defaults["page"]?.ToString()?.Trim('/')
```

Comment:
You can use `TryGetValue()` to avoid two operations for `"page"`.

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/3338/commits/b46f36e91b0b0353f2532c5a57ce357f339398ed